### DISCUSSION TITLE
Failure of downloading after the newest update in older version of MacOS

### DISCUSSION THREAD

**Discussion by EICr** [aclap-dev/video-downloadhelper#860] [Help and Q&A]
### Your question

Hi there, since your addon's update I cannot download any video via firefox addons (did not try other browsers but I guess it is not a browser problem).

The video links I tried are here:
https://www.youtube.com/watch?v=6bNOnXTe4Ok
https://www.hsbc.com/corners (the videos in that website)

Error information is here:

Exit code: null dyld: cannot load 'ffprobe' (load command 0x80000034 is unknown)

moz-extension://54803ac0-4d8e-4916-a396-c3d7bd9bbfbd/background/main.js:1
receive@moz-extension://54803ac0-4d8e-4916-a396-c3d7bd9bbfbd/background/main.js:1:13152
fd</callCatchAppNotFound/</<@moz-extension://54803ac0-4d8e-4916-a396-c3d7bd9bbfbd/background/main.js:1:42659

I guess it is not a new problem because I found this error in other forum (not with this addon) and in this forum (but still no answers), really hope that your helper would not ignore some people are still using older operating system version. Thank you so much.

By the way, sincere thanks for your addon for past two years, it helps a lot. So I really wish that you could help.

### Addon & CoApp version and details

version: 9.0.2.3
channel: stable
build date: 2024-05-27
build options: linuxlic: false, noyt: false, browser: firefox
lang: zh-CN
license: unset
platform: x86-64 mac
UA: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.14; rv:109.0) Gecko/20100101 Firefox/115.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "/Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "mac",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/Users/eicr",
  "converterError": "Exit without answer"
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

I am running Windows 11 Pro 64-bit version 23H2 build 22631.3737, Firefox 127.0 64-bit, licensed VDH 9.0.2.3a1 beta, CoApp 2.0.19.  That includes new Windows maintenance Microsoft pushed out just today as well as a new Firefox also pushed out just today.

Your YouTube video downloads just fine for me.  I didn't check the HSBC web site.  Maybe you should try the beta.  Click the link at the top of any page of this forum labelled Wiki.  That's a user manual of sorts.  The chapters are listed down the right side of the page.  Click the one that mentions beta versions.  You'll find the beta there.

**Comment by kidslow** [comment]
I would classify this as a bug. Here is the workaround (assuming you have ffmpeg installed via homebrew to /usr/local/bin), using terminal:

**cd /Applications/net.downloadhelper.coapp.app/Contents/MacOS/
sudo mv ffmpeg ffmpeg.dist
sudo mv ffprobe ffprobe.dist
sudo ln -s /usr/local/bin/ffmpeg ffmpeg
sudo ln -s /usr/local/bin/ffprobe ffprobe**

If you're feeling frisky, rm the distributed ffmpeg and ffprobe rather than renaming. I'm not sure why the developers of coapp would find it necessary to install the binaries and ignore the system version of those, but I would say what they are doing is improper and could be solved with some more logic to detect if there are system versions of those binaries in /usr/bin /usr/local/bin or /opt/bin. 

I solved problems resulting in the exact same error using VDH on 10.14.6 using Firefox 115.4 with this solution. Be warned that Homebrew also no longer supports Mojave, and you will pull your hair out if you try to use it, so macports is a better choice if you are starting from scratch. Just FYI. Just make sure there is an ffmpeg executable or link to one in /usr/local/bin and my above solution will work.

**Reply by GrampaWildWilly** [reply]
I do not work for VDH.  I am just another user.

I know diddly squat about Linux & just as much about Mac.  But there are versions of the CoApp that don't include ffmpeg.  Don't ask me how to install & use them.  I just know they exist.  At the top of any page of this forum, click the link for aclap-dev.  That takes you to a page where you can find a link for the CoApp.  Click that.  That takes you to a page where there is a link on the right side for Releases.  Click that.  On that page you might have to click a link marked `Assets` & you might have to click another link marked `Show all xx assets`.  You're on your own after that.  I can't say if any of that applies to Mac.  I know there is a version of Linux inside the Mac OS but I can't tell you what contortions you have to go through to do Linux-y things on Mac.

There is a version of ffmpeg packaged with the CoApp because @paulrouget Paul made some local modifications to ffmpeg to get around some restrictions that some users were encountering.  I can't tell you specifically what those restrictions & local modifications are.  I just know Paul did that.  So you shouldn't be so quick to judge what is & is not proper.

**Reply by kidslow** [reply]
Thanks for the suggestions, but I don't see any Mac package for x86-64 that does not include the binary for ffmpeg and ffprobe, but I followed your directions and only checked the latest download I could see --> vdhcoapp-mac13-x86_64-installer.pkg

Embedding common binaries inside the package is a kludge. Maybe improper was not the proper choice of words, would you accept inconvenient? Perhaps could be done in a more universal way to not cause problems, or offer a configuration override that allows a local user to point at system binaries for ffmpeg and ffprobe and bypass the embedded ones.
